projects/storefrontlib/src/lib/pwa/services/add-to-home-screen.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { PWAModuleConfig } from '../pwa.module-config';
+import { GlobalMessageType } from './../../global-message/models/message.model';
+import { GlobalMessageService } from '../../global-message/facade/global-message.service';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable()
+export class AddToHomeScreenService {
+  private deferredEvent: any;
+
+  private canPrompt = new BehaviorSubject<boolean>(false);
+
+  canPrompt$ = this.canPrompt.asObservable();
```

Comment:
why we want an observable canPrompt$? A boolean variable is enough, right?

Reply:
Boolean isn't enough. Each consumer should react instantly when `beforeinstallprompt` event is fired. Button in a mobile menu is initialized on app start. The event can be triggered 30 second later and this component should be shown there right after.